.github/workflows/tag_release.yaml

```diff
@@ -0,0 +1,75 @@
+name: "Create Tag on Pull Request merge"
+on:
+  pull_request:
+    types:
+      - closed
+    branches:
+      - main
+
+jobs:
+  TagMerge:
+    name: Tag on Pull Request merge
+    permissions:
+      issues: write
+      pull-requests: write
+      contents: write
+
+    runs-on: ubuntu-latest
+    if: github.event.pull_request.merged == true
+    steps:
+      - name: Generate Tag
+        id: tagged
+        run: |
+          echo $(curl \
+            --silent \
+            -H "Accept: application/vnd.github+json" \
+            -H "Authorization: Bearer ${{ secrets.GITHUB_TOKEN }}"\
+            "https://api.github.com/repos/${{ github.repository }}/tags" \
+            | jq -r '.[].name |= sub("-rc0|v"; "";"g") | select(length > 1) | max_by(.name) | .name' \
+            | awk '
+            function inc(s){
+              split(s, a, ".")
+              a[3]++
+              if (a[3]>=10){
+                a[2]++;a[3]=0
+              }
+              if (a[2]>=10){
+                a[1]++;a[2]=0
+              }
+              return a[1]"."a[2]"."a[3]
+
+            }
+            { print "newtag="inc($1) }') >> $GITHUB_OUTPUT
```

Comment:
Mega chad tag incrementer :sunglasses: 

One thing I just thought to keep the actions file clean is to drop this type of thing into the `Makefile` or a `bin/generate-tag.sh` so that you can call it here with `make tag` or something?